cmd/subctl/recover_broker_info.go

```diff
@@ -0,0 +1,172 @@
+	"fmt"
+	"github.com/submariner-io/submariner-operator/api/v1alpha1"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/subctl/internal/cli"
+	"github.com/submariner-io/subctl/internal/constants"
+	"github.com/submariner-io/subctl/internal/exit"
+	"github.com/submariner-io/subctl/internal/restconfig"
+	"github.com/submariner-io/subctl/pkg/broker"
+	"github.com/submariner-io/subctl/pkg/cluster"
+)
+
+var recoverRestConfigProducer = restconfig.NewProducer().WithPrefixedContext("broker").
+	WithDefaultNamespace(constants.DefaultBrokerNamespace).
+	WithPrefixedNamespace("broker", constants.DefaultBrokerNamespace)
+
+// recoverBrokerInfo represents the reconstruct command.
+var recoverBrokerInfo = &cobra.Command{
+	Use:   "recover-broker-info",
+	Short: "Recovers the broker-info.subm file from the installed Broker",
+	Run: func(cmd *cobra.Command, args []string) {
+		exit.OnError(recoverBroker(cli.NewReporter()))
+	},
+}
+
+func init() {
+	recoverRestConfigProducer.SetupFlags(recoverBrokerInfo.Flags())
+	rootCmd.AddCommand(recoverBrokerInfo)
+}
+
+func recoverBroker(status reporter.Interface) error {
+	// if --brokerconfig flag provided, get the broker and proceed to get Submariner
+	contextFound, err := recoverRestConfigProducer.RunOnSelectedPrefixedContext("broker",
+		func(brokerCluster *cluster.Info, brokerNamespace string, status reporter.Interface) error {
+			brokerObj, err := getBroker(brokerCluster, brokerNamespace, "Please try again with another cluster", status)
+			if err != nil {
+				return err
+			}
+
+			clusters, err := brokerCluster.GetJoinedClusters(brokerNamespace)
+			if err != nil {
+				return status.Error(err, "error listing joined clusters")
+			}
+
+			ok, err := tryToRecoverFromBroker(brokerCluster, brokerObj, brokerNamespace, clusters, status)
+			if ok || err != nil {
+				return err
+			}
+
+			status.Warning("Submariner is not installed on the same cluster as Broker")
+			status.Start("Checking if Submariner is installed on a different cluster")
+			return recoverRestConfigProducer.RunOnSelectedContext(
+				func(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
+					if isSubmJoinedToBroker(clusters, submCluster) {
+						status.Success("Found Submariner on cluster %s", submCluster.Name)
+						return broker.RecoverData(brokerCluster, submCluster, brokerObj, namespace, status)
+					}
+
+					return status.Error(
+						errors.New(fmt.Sprintf("submariner is not installed on this cluster %s. "+
+							"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
+							"", submCluster.Name)), "")
+				}, status)
+		}, status)
+
+	// if --brokerconfig not provided, search for broker on current context and proceed to get Submariner
+	if !contextFound && err == nil {
```

Comment:
Again, return early here to reduce indentation.

```suggestion
        if contextFound || err != nil {
            return err
        }
```